sympy/polys/polytools.py

```diff
@@ -6831,6 +6899,9 @@ def all_roots(f, multiple=True, radicals=True):
     """
     try:
         F = Poly(f, greedy=False)
+        if extension and not F.domain.is_AlgebraicField:
+            F = Poly(F.expr, extension=True)
```

Comment:
Do we need `greedy=False` here?

The effect of greedy is that it means that numbers like `sqrt(2)` are considered coefficients rather than generators:
```python
In [11]: Poly(x + sqrt(2))
Out[11]: Poly(x + (sqrt(2)), x, sqrt(2), domain='ZZ')

In [12]: Poly(x + sqrt(2), greedy=False)
Out[12]: Poly(x + sqrt(2), x, domain='EX')
```
I don't think that matters here though because we should reject multivariate polynomials and also polynomials whose domain is not QQ or (if extension=True) AlgebraicField.

I think this should just be:
```python
F = poly(f, extension=extension)
```

Reply:
I just tried this but it causes some conflict in polyoptions. 

```
>>> Poly(x + sqrt(2), x, extension=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/maaz/Documents/sympy/sympy/polys/polytools.py", line 168, in __new__
    opt = options.build_options(gens, args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/maaz/Documents/sympy/sympy/polys/polyoptions.py", line 742, in build_options
    return Options(gens, args)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/maaz/Documents/sympy/sympy/polys/polyoptions.py", line 153, in __init__
    preprocess_options(args)
  File "/Users/maaz/Documents/sympy/sympy/polys/polyoptions.py", line 151, in preprocess_options
    self[option] = cls.preprocess(value)
                   ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/maaz/Documents/sympy/sympy/polys/polyoptions.py", line 552, in preprocess
    raise OptionError("'False' is an invalid argument for 'extension'")
sympy.polys.polyerrors.OptionError: 'False' is an invalid argument for 'extension'
```

In light of that, I think it makes sense to keep the `greedy=False` option which is as before?

Reply:
The polyoptions stuff is a bit over-engineered and complicates simple things. I guess we can keep greedy. We shouldn't construct the Poly twice though.

Reply:
Sure